### ISSUE TITLE
No licence

### ISSUE THREAD

**Issue by IsaiahPatton** [CodeCrafter47/Rainbow#5]
theres no licence.txt file

**Comment by CodeCrafter47** [comment]
I agree that having no license is kind of bad. I'll see whether I can change that.

**Comment by Bayside308** [comment]
Are you still looking into adding a license?

**Comment by CodeCrafter47** [comment]
Yes, however a lot of the code has been written by @DoingItWell which means that without his consent we can't change the license. So there won't be any progress on this issue until he shows up.

**Comment by gdude2002** [comment]
A license is always good. Ticket motivation is interesting, though.

You're probably aware, but just in case you're not, the person that opened this ticket is working on an uncelebrated fork of this project using a Spigot-like patching system, [here](https://github.com/RainbowPlus/RainbowPlus-Server) ([archive.is](http://archive.is/sCv30)). It took me a while to figure out what was actually happening in that repo as there doesn't appear to be any attribution or acknowledgement of it being based off of this project whatsoever.

Since there are JARs available for distribution, and this repository does not have any license applied to it, Isaiah's distribution violates the contributors' rights on this project (No license on a repo means that one is not allowed to distribute it in any way).

To add insult to injury - Isiah's project is accepting donations already, as can be seen on [the project website](https://rainbowplus.github.io/) ([archive.is](http://archive.is/gxMNF)). This is despite being a relatively small, new project, which.. doesn't really offer that much over Rainbow as it stands, and which has almost all of its functionality provided by Rainbow itself.

Normally I wouldn't just barge in like this, but it seems that Isaiah here decided to spam [one of my orgs](https://github.com/GlowstoneMC) by inviting all of our public members to his own org, which is obviously not OK and fairly annoying. Plus, I figured you guys should be aware of stuff like this.

I have taken backups of both repos if they disappear and you need a copy of them for whatever reason.

**Comment by Bayside308** [comment]
Hey gdude2002,

I believe CodeCrafter is aware of Isiah's fork, as he often makes pull requests to Rainbow from features he added to Rainbow++. Not sure how DiW would feel though, and you do make a good point with the whole distributions issue (no license = all rights reserved, correct?).

You also point out donations, which, to be fair, [don't seem to be accepted yet](http://prnt.sc/e1ntd8). Again, not sure how CodeCrafter feels (and I'm an insignificant member of the Rainbow community).

**Comment by IsaiahPatton** [comment]
You might what to be on the right website befour you say I'm taking donations.

**Comment by DaMatrix** [comment]
Do you not know what archive.is is?

**Comment by CodeCrafter47** [comment]
I'm aware of Isaiah's fork and I have no objections to this project being forked, after all I intend to put the code in this repository under a free software license.

Isaiah inviting all members of your organization to his is not OK, as you said.

Regarding the licensing issue itself, I can't do anything about it until DiW comes back online, so we probably have to live with the way it is now for some time.

Regarding donations, we do not take any donations, but I don't have a problem with others trying to get some money for what they're doing.

And @Bayside308 I don't think you are an **insignificant** member of our community.

**Comment by gdude2002** [comment]
@IsaiahPatton If you don't intend on accepting donations, then you should:

* Link to the correct site in the README (I'm assuming the forumotion site is the correct one?)
    * Remove the "Donate" link in the hamburger menu as well
* Hide, remove or edit your `github.io` site repo

I would advise linking directly to this repo in the README as well, to avoid confusion

@CodeCrafter47 [This fork](https://github.com/IsaiahPatton/Rainbow) is not what this ticket is about.

Also, I believe most communities are accepting of implicit consent by users that don't respond to requests for it for 30 days or more.

@Bayside308 He would be unable to make pull requests from the repo I linked as it is not marked as a fork of this one.

**Comment by CodeCrafter47** [comment]
@gdude2002 I'm fully aware of Isaiah's RainbowPlus project. I think it was a real fork (the way Github uses the term) at the beginning, but now uses that patch based system.

Regarding the license, @DoingItWell is the original creator of Rainbow, the project still uses a lot of the code he wrote, and I feel it would be inappropriate to *change* the license of the project without asking him, even though I don't think he would have any objections.

Also Rainbow is a very small project with probably just about a dozen of users, so I don't think it's worth worrying too much about all of this.
